django_airavata/apps/api/views.py

```diff
@@ -1520,21 +1523,56 @@ def move_file(file_path, file_name, file_type):
             data_product = tus.move_tus_upload(uploadURL, move_file)
         return self._create_response(request, path, uploaded=data_product)
 
+    # Accept wither to replace file or to replace file content text.
+    def put(self, request, path="/", format=None):
+        # Replace the file if the request has a file upload.
+        if 'file' in request.FILES:
+            self.delete(request=request, path=path, format=format)
```

Comment:
Can we have a data_products_helper.replace_data_product(request, data_product, file) instead of deleting and recreating? Reason is that we may want to re-register the DataProduct in the future and best to keep that logic together.

Reply:
I didn't want to do so since I could use `delete` and `post` endpoints methods and take the advantage of its validations here.

Reply:
Okay, yeah I think that's fine.